src/box/lua/config/configdata.lua

```diff
@@ -5,11 +5,18 @@
 
 local fio = require('fio')
 local fun = require('fun')
+local json = require('json')
 local urilib = require('uri')
 local digest = require('digest')
+local xlog = require('xlog')
 local instance_config = require('internal.config.instance_config')
 local cluster_config = require('internal.config.cluster_config')
 
+-- The map of peers, which don't have names set. Global, as it should
+-- not change during fetching new config data. Created only during
+-- on first startup.
+local missing_names = {}
```

Comment:
I don't like the fact, that this requires using "global" variable. But the module is supposed to "accumulate configuration data" and missing names are definitely this data, so it seems like names it's a right place for names checking. @Totktonada, what do you think?

Reply:
AFAIU, this map is deduced from two sources: from a snapshot and from a config.

Later, I guess, it is changed with changing of the database state (`_cluster` space).

The quoted rule means that the configdata object is primarily just a view of a particular cluster config (in context of a particular instance).

The idea is that if the object has such a simple contract -- create it and read a needed data -- it makes it easier to understand and use.

I agree that `missing_names` doesn't work this way. Let's think where it can reside.

My first thought is whether we can expose some (existing) `:names()` from a module that represents a snapshot data in a format that allows a simple comparison against configdata's `:names()` and `:peers()`. If so, we can compare them on demand in the `box_cfg` applier.

Or we can expose existing names from a snapshot data module and expose `:missing_names()` from configdata to compare the snapshot data with the config on demand. It makes configdata dependent from a database state, but it doesn't look critical.

I don't mind if you'll keep it as is for a while, though.

I'll return here back if I'll have more thoughts after reading next patches.

Reply:
> Or we can expose existing names from a snapshot data module and expose :missing_names() from configdata to compare the snapshot data with the config on demand. It makes configdata dependent from a database state, but it doesn't look critical.

I like that, done